go/logic/applier.go

```diff
@@ -568,12 +568,17 @@ func (this *Applier) StartSlaveSQLThread() error {
 
 // StopReplication is used by `--test-on-replica` and stops replication.
 func (this *Applier) StopReplication() error {
-	if err := this.StopSlaveIOThread(); err != nil {
-		return err
-	}
-	if err := this.StopSlaveSQLThread(); err != nil {
-		return err
+	if this.migrationContext.TestOnReplicaSkipReplicaStop {
```

Comment:
Let me know if I'm getting too bossy.
I'd rather do the `if this.migrationContext.TestOnReplicaSkipReplicaStop` externally to this function. So I would rather have the logic in `migrator.go` instead.